pkg/runcfanotify/runcfanotify.go

```diff
@@ -135,7 +135,12 @@ func (n *RuncNotifier) watchPidFileIterate(pidFileDirNotify *fanotify.NotifyFD,
 	}
 
 	// Don't leak the fd received by GetEvent
-	defer data.Close()
+	// Cannot use data.Close() and data.File() for the same data: the file
+	// would be closed twice:
+	// 1. data.Close() -> closed directly with unix.Close()
+	// 2. data.File()  -> closed indirectly by os.File's finalizer
+	dataFile := data.File()
+	defer dataFile.Close()
```

Comment:
Just to be sure I understood. This is not strictly required as the GC will take care anyway when dataFile goes out of scope?

Reply:
That's true... we could probably write:
```
_ = data.File()
```
And then the GC will close it at some point.

But I would argue it's bad to keep unknown fds for an unknown amount of time. For examples, pipe files can be open and closed as a synchronisation mechanism between processes. If we wait before closing it, I wonder if it could delay an application for an unreasonable amount of time.

Reply:
You also don't want open fds that you are not using. Open fds is a scarce resource. You always want to close whenever you can :)